components/searchable_channel_list.jsx

```diff
@@ -204,15 +221,54 @@ export default class SearchableChannelList extends React.Component {
             );
         }
 
+        let channelDropdown;
+
+        if (this.props.canShowArchivedChannels) {
+            channelDropdown = (
+                <MenuWrapper
+                    id='ChannelsMoreDropdown'
+                    className='more-modal__dropdown'
+                >
+                    <a>
+                        <span>{this.props.shouldShowArchivedChannels ? localizeMessage('more_channels.show_archived_channels', 'Show: Archived Channels') : localizeMessage('more_channels.show_public_channels', 'Show: Public Channels')}</span>
+                        <span className='caret'/>
+                    </a>
+                    <Menu
+                        openLeft={false}
+                        ariaLabel={localizeMessage('team_members_dropdown.menuAriaLabel', 'Team member role change')}
+                    >
+                        <Menu.ItemAction
+                            id='ChannelsMoreDropdownPublic'
+                            onClick={() => {
+                                this.props.toggleArchivedChannels(false);
+                            }}
```

Comment:
Can you make this into a method of the `SearchableChannelList` so that it doesn't cause the `Menu.ItemAction` to re-render each time because it's being passed a new function?